synapse/storage/databases/main/roommember.py

```diff
@@ -314,7 +314,7 @@ async def get_rooms_for_local_user_where_membership_is(
             The RoomsForUser that the user matches the membership types.
         """
         if not membership_list:
-            return None
+            return []
```

Comment:
Nothing passes in an empty list so this should be a no-op change. Doing this allows us to avoid unwrapping the optional, plus I think makes semantic sense.